GVFS/GVFS.Common/Tracing/EventListener.cs

```diff
@@ -14,30 +14,27 @@ protected EventListener(EventLevel maxVerbosity, Keywords keywordFilter)
             this.keywordFilter = keywordFilter;
         }
 
+        public event EventHandler ListenerRecovery;
+
+        public event EventHandler<string> ListenerFailure;
+
         public virtual void Dispose()
         {
         }
 
-        public bool? TryRecordMessage(TraceEventMessage message, out string errorMessage)
+        public void RecordMessage(TraceEventMessage message)

```

Comment:
Because the posting of events over the pipe is now disconnected from actual tracer message write, failure (and recovery) is now communicated using events.